### PR TITLE
fix: :zap: Optimize `GetPosition` class

### PR DESCRIPTION
**Pull request by ujas-m-simformsolutions**
# Description

Reduced call for render object. Improved conditions.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have followed the [Contributor Guide] when preparing my PR.
- [ ] I have updated/added tests for ALL new/updated/fixed functionality.
- [ ] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [ ] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

None.

[Contributor Guide]: https://github.com/SimformSolutionsPvtLtd/flutter_showcaseview/blob/master/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

File: lib/src/get_position.dart [hunk 1]
```diff
@@ -23,75 +23,87 @@
...
+    }
+  }
 
-    var boxOffset = box.localToGlobal(const Offset(0.0, 0.0));
-    if (boxOffset.dx.isNaN || boxOffset.dy.isNaN) {
-      return const Rect.fromLTRB(0, 0, 0, 0);
+  Rect getRect() {
+    if (_box == null ||
+        _boxOffset == null ||
+        (_boxOffset?.dx.isNaN ?? true) ||
```
Comment (aditya-css) [inline lib/src/get_position.dart:54]: Please create an extension for this.

File: lib/src/get_position.dart [hunk 2]
```diff
@@ -23,75 +23,87 @@
...
   }
 
   ///Get the top position of the widget
   double getTop() {
-    final box = key!.currentContext!.findRenderObject() as RenderBox;
-    final boxOffset = box.localToGlobal(const Offset(0.0, 0.0));
-    if (boxOffset.dy.isNaN) return 0 - padding.top;
-    final topLeft = box.size.topLeft(boxOffset);
+    if (_box == null || _boxOffset == null || (_boxOffset?.dy.isNaN ?? true)) {
+      return 0 - padding.top;
```
Comment (aditya-css) [inline lib/src/get_position.dart:86]: I believe you can skip subtracting from 0 and directly assign negative sign here.

**Review by aditya-css — CHANGES_REQUESTED** [review]

**Review by aditya-css — APPROVED** [review]

**Comment by guenth39** [comment]
When will this be released? I need the fix and therefore have to use master right now, it would be nice to be able to switch back to the pub version.

**Comment by aditya-css** [comment]
Hey @guenth39, We would like to inform that we have released this commit with 2.1.0 version of this package.

**Pull request merged**